aws/resource_aws_ec2_transit_gateway_peering_attachment_test.go

```diff
@@ -0,0 +1,413 @@
+			continue
+		}
+
+		peeringAttachment, err := ec2DescribeTransitGatewayPeeringAttachment(conn, rs.Primary.ID)
+
+		if isAWSErr(err, "InvalidTransitGatewayAttachmentID.NotFound", "") {
+			continue
+		}
+
+		if err != nil {
+			return err
+		}
+
+		if peeringAttachment == nil {
+			continue
+		}
+
+		if aws.StringValue(peeringAttachment.State) != ec2.TransitGatewayAttachmentStateDeleted {
+			return fmt.Errorf("EC2 Transit Gateway Peering Attachment (%s) still exists in non-deleted (%s) state", rs.Primary.ID, aws.StringValue(peeringAttachment.State))
+		}
+	}
+
+	return nil
+}
+
+func testAccCheckAWSEc2TransitGatewayPeeringAttachmentDisappears(transitGatewayPeeringAttachment *ec2.TransitGatewayPeeringAttachment) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		conn := testAccProvider.Meta().(*AWSClient).ec2conn
+
+		input := &ec2.DeleteTransitGatewayPeeringAttachmentInput{
+			TransitGatewayAttachmentId: transitGatewayPeeringAttachment.TransitGatewayAttachmentId,
+		}
+
+		if _, err := conn.DeleteTransitGatewayPeeringAttachment(input); err != nil {
+			return err
+		}
+
+		return waitForEc2TransitGatewayPeeringAttachmentDeletion(conn, aws.StringValue(transitGatewayPeeringAttachment.TransitGatewayAttachmentId))
+	}
+}
+
+func testAccAWSEc2TransitGatewayPeeringAttachmentConfig_base(rName string) string {
+	return fmt.Sprintf(`
+resource "aws_ec2_transit_gateway" "test" {
+  tags = {
+    Name = %[1]q
+  }
+}
+
+resource "aws_ec2_transit_gateway" "peer" {
+  provider = "aws.alternate"
+
+  tags = {
+    Name = %[1]q
+  }
+}
+`, rName)
+}
+
+func testAccAWSEc2TransitGatewayPeeringAttachmentConfig_sameAccount_base(rName string) string {
+	return testAccAlternateRegionProviderConfig() + testAccAWSEc2TransitGatewayPeeringAttachmentConfig_base(rName)
+}
+
+func testAccAWSEc2TransitGatewayPeeringAttachmentConfig_differentAccount_base(rName string) string {
+	return testAccAlternateAccountAlternateRegionProviderConfig() +
+		testAccAWSEc2TransitGatewayPeeringAttachmentConfig_base(rName) +
+		fmt.Sprintf(`
+data "aws_caller_identity" "peer" {
+  provider = "aws.alternate"
+}
+`)
+}
+
+func testAccAWSEc2TransitGatewayPeeringAttachmentConfigBasic_sameAccount(rName string) string {
+	return testAccAWSEc2TransitGatewayPeeringAttachmentConfig_sameAccount_base(rName) + fmt.Sprintf(`
+resource "aws_ec2_transit_gateway_peering_attachment" "test" {
+  peer_region             = %[1]q
+  peer_transit_gateway_id = "${aws_ec2_transit_gateway.peer.id}"
+  transit_gateway_id      = "${aws_ec2_transit_gateway.test.id}"
+}
+`, testAccGetAlternateRegion())
+}
+
+func testAccAWSEc2TransitGatewayPeeringAttachmentConfigBasic_differentAccount(rName string) string {
+	return testAccAWSEc2TransitGatewayPeeringAttachmentConfig_differentAccount_base(rName) + fmt.Sprintf(`
+resource "aws_ec2_transit_gateway_peering_attachment" "test" {
+  peer_account_id         = "${data.aws_caller_identity.peer.account_id}"
```

Comment:
Nit: This can be simplified to use the `owner_id` attribute of the `aws_ec2_transit_gateway` resource 👍 

```suggestion
  peer_account_id         = "${aws_ec2_transit_gateway.peer.owner_id}"
```